src/main/java/org/openmrs/addonindex/backend/Bintray.java

```diff
@@ -65,7 +69,36 @@ public AddOnInfoAndVersions getInfoAndVersionsFor(AddOnToIndex addOnToIndex) thr
 			return handlePackageJson(addOnToIndex, json);
 		}
 	}
-	
+
+	@Override
+	public void fetchDownloadCounts(AddOnToIndex toIndex, AddOnInfoAndVersions infoAndVersions) throws Exception {
+		//We won't be checking for existing info as download counts is dynamic
+		// and has to be fetched each time
+		logger.info("Fetching Download Counts for " + toIndex.getUid());
+		BintrayDownloadCounts downloadCounts;
+		String json;
+		BintrayPackageDetails bintrayPackageDetails = toIndex.getBintrayPackageDetails();
+		String baseurl = "https://bintray.com/statistics/packageGeoStats?&pkgPath=/" + bintrayPackageDetails.getOwner()
+				+ "/" + bintrayPackageDetails.getRepo() + "/" + bintrayPackageDetails.getPackageName();
+		try {
+			json = restTemplateBuilder.build().getForObject(baseurl, String.class);
+			downloadCounts = mapper.readValue(json, BintrayDownloadCounts.class);
+			Integer totalDownloadCounts = 0;
+			for (Integer downloadCount : downloadCounts.getTotalDownloads().values()) {
+				try {
+					totalDownloadCounts += downloadCount;
+				}
+				catch (Exception e) {
+					logger.error("Error getting details for " + toIndex.getUid(), e);
+				}
+			}
+			infoAndVersions.setDownloadCounts(totalDownloadCounts);
+		}
+		catch (Exception ex) {
+			throw new RuntimeException("Download counts from " + baseurl + " could not be parsed", ex);
```

Comment:
Never catch a checked exception and rethrow it as an unchecked RuntimeException. Yes, this seems convenient to do sometimes, but it's never correct.
Also in this specific case our method is _allowed_ to throw Exception, so you can throw this as a plain Exception, or just let it bubble up.